presto-main/src/main/java/io/prestosql/sql/planner/QueryPlanner.java

```diff
@@ -797,6 +797,38 @@ public static Expression coerceIfNecessary(Analysis analysis, Expression origina
                 analysis.isTypeOnlyCoercion(original));
     }
 
+    public static NodeAndMappings coerce(PlanNode node, List<Symbol> fields, List<Type> types, SymbolAllocator symbolAllocator, PlanNodeIdAllocator idAllocator)
```

Comment:
preserve `processAndCoerceIfNecessary` name. It was more accurate.

Reply:
"process" is not accurate. This just adds coercions (or identity projections if coercions are not needed)

Reply:
Actually, no, I won't do that. The reason I went with this name is that there's already another method called coerceIfNecessary in RelationPlanner that conflicts with this one and requires qualifying the invocation with the class name